public/app/features/alerting/unified/utils/rulerClient.ts

```diff
@@ -81,7 +81,11 @@ export function getRulerClient(rulerConfig: RulerDataSourceConfig): RulerClient
   };
 
   const deleteRule = async (ruleWithLocation: RuleWithLocation): Promise<void> => {
-    const { namespace, group, rule } = ruleWithLocation;
+    let { namespace, group, rule } = ruleWithLocation;
+
+    if (isGrafanaRulerRule(rule)) {
```

Comment:
Instead of overwriting the namespace here, would it make more sense to add the `namespace_uid` / `folder_uid` to the `ruleWithLocation` interface and update the code for [`findEditableRule`](https://github.com/grafana/grafana/blob/577b3f2fb20f189c95d806d5ef5f016f35097933/public/app/features/alerting/unified/utils/rulerClient.ts#L36-L45)?

Reply:
Good idea, done!